NetworkConfigurations/NetworkAddressingAndFilteringSubCategories/Stateful-NAT.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Stateful Network Address Translation
+grand_parent: Network Configuration
+parent: Network Addressing and Filtering
+nav_order: 5
+layout: default
+---
+
+# Stateful Network Address Translation (NAT)
+
+Stateful Network Address Translation (NAT), also known as dynamic NAT, facilitates the mapping of multiple private IP addresses to public addresses.
+
+- It was introduced in the [Beeblebrox Release (v2)](https://github.com/dentproject/dentOS/releases/tag/v2.0) of DENT.
+
+Configuration of stateful NAT involves employing the `tc iproute2` tool alongside the `tc-ct` action. For dynamic NAT entry offloading, integrate the `tc-ct` action with the flower filter rule and ingress qdisc.
```

Comment:
I do not think these are right. Or at least I do not see these actions used in the doc sections below. Could you please double check?

Reply:
I'm not sure about the release here, but this is something mentioned in the DENT roadmap. I can change that if you provide some info or simply remove this line.

Regarding the usage of commands, the use of `tc` (iproute2) and `tc-ct` has been documented. Example commands used:

```
tc filter add dev swp20 ingress proto ip pref 2 flower ct_state -trk action ct
tc filter add dev swp21 egress prio 10 proto ip flower action ct commit nat src addr 91.245.77.1 pipe action pass
```

Reply:
Let's not delete these just simply verify this config works. Please add a graph of sample topology (you can use VMs) and verify the config works before deleting/modifying anything.